### PR TITLE
Add tvm

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Checklist

- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vend other packages
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
Hi @tqchen , Could you please help me here to create a package on ``conda-forge`` for ``tvm``?

File: recipes/tvm/build.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	-DCMAKE_BUILD_TYPE=Release \
+	-DCMAKE_CONFIGURATION_TYPES="Release" \
+	-DUSE_CUDA=OFF \
+	-DCMAKE_INSTALL_PREFIX=${PREFIX} \
+ 	-DCMAKE_PREFIX_PATH=${PREFIX} \
+        -DDLPACK_PATH=${PREFIX}/include/dlpack \
+	-DDMLC_CORE_PATH=${PREFIX}/lib \
+	-DUSE_VULKAN=OFF \
+	-DUSE_LLVM=$PREFIX/bin/llvm-config \
+	-DCMAKE_TOOLCHAIN_FILE=${RECIPE_DIR}/cross_compile.cmake \
```
Comment (isuruf) [inline recipes/tvm/build.sh]: You can remove this. In my experience, it causes more issues than it solves.

File: recipes/tvm/cmake_patch.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
++include_directories("${CMAKE_PREFIX_PATH}/include")
++include_directories("${CMAKE_PREFIX_PATH}/include/dlpack")
++include_directories("${CMAKE_PREFIX_PATH}/include/dmlc")
++include_directories("${CMAKE_PREFIX_PATH}/include/ir")
++include_directories("${CMAKE_PREFIX_PATH}/include/tvm")
++include_directories("${CMAKE_PREFIX_PATH}/include/tvm/node")
++include_directories("${CMAKE_PREFIX_PATH}/include/arithmetic")
++include_directories("${CMAKE_PREFIX_PATH}/include/base")
++include_directories("${CMAKE_PREFIX_PATH}/include/compiler-rt")
+ include_directories("3rdparty/compiler-rt")
```
Comment (isuruf) [inline recipes/tvm/cmake_patch.patch]: `compiler-rt` is still vendored?

File: recipes/tvm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - compiler-rt
+    - cmake
+    - clangdev
+    - llvmdev
```
Comment (isuruf) [inline recipes/tvm/meta.yaml]: `clangdev`, `llvmdev`, `compiler-rt` should be in `host` as they are used as libraries.

File: recipes/tvm/cmake_patch.patch (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/tvm/cmake_patch.patch]: It needs this header https://github.com/dmlc/tvm/blob/v0.5/3rdparty/compiler-rt/builtin_fp16.h
but it seems that it is not being packaged by ``compiler-rt``, should I raise it on the feedstock?

**Comment by marcelotrevisani** [comment]
@conda-forge/help-c-cpp 
Does anyone have any ideas here?

**Pull request closed without merging**